src/libertem/analysis/com.py

```diff
@@ -200,35 +200,58 @@ def get_generic_results(self, img_sum, img_y, img_x):
             y_real, y_imag = np.real(y_centers), np.imag(y_centers)
 
             return COMResultSet([
-                AnalysisResult(raw_data=x_real, visualized=visualize_simple(x_real),
+                AnalysisResult(raw_data=x_real, visualized=visualize_simple(x_real, damage=damage),
                        key="x_real", title="x [real]", desc="x component of the center"),
-                AnalysisResult(raw_data=y_real, visualized=visualize_simple(y_real),
+                AnalysisResult(raw_data=y_real, visualized=visualize_simple(y_real, damage=damage),
                        key="y_real", title="y [real]", desc="y component of the center"),
-                AnalysisResult(raw_data=x_imag, visualized=visualize_simple(x_imag),
+                AnalysisResult(raw_data=x_imag, visualized=visualize_simple(x_imag, damage=damage),
                        key="x_imag", title="x [imag]", desc="x component of the center"),
-                AnalysisResult(raw_data=y_imag, visualized=visualize_simple(y_imag),
+                AnalysisResult(raw_data=y_imag, visualized=visualize_simple(y_imag, damage=damage),
                        key="y_imag", title="y [imag]", desc="y component of the center"),
             ])
         else:
-            f = CMAP_CIRCULAR_DEFAULT.rgb_from_vector((x_centers, y_centers, 0))
+            damage = damage & np.isfinite(x_centers) & np.isfinite(y_centers)
+            if np.count_nonzero(damage) > 0:
+                vmax = np.sqrt(np.max(x_centers[damage]**2 + y_centers[damage]**2))
+            else:
+                vmax = 1
+            f = CMAP_CIRCULAR_DEFAULT.rgb_from_vector((x_centers, y_centers, 0), vmax=vmax)
```

Comment:
Why do we need a `vmax` here now? I guess because `rgb_from_vector` doesn't like an empty array as input?

Reply:
It _could_ happen that `damage` is empty for some reason, for example if the input data contains NaN. I wrote the code when I istarted introducing `damage` and it wasn't always set properly. IIRC an empty `damage` trips the calculation of `vmax`. If it is OK I'd prefer to leave it as-is, it is probably safer.

Reply:
Sounds good to keep as-is, maybe add a small comment?

Reply:
Not sure how other analyses, plotting etc react to input data with NaN. It might trip for similar reasons. Would it be OK to leave that to later if/when it actually creates an issue? At that point one could introduce an all-NaN dataset to the test circuit.

Reply:
Ok! Most analyses right now use `visualize_simple`, which should properly ignore those NaNs.